File: DigitalScrumBoard1/Models/Notification.cs
namespace DigitalScrumBoard1.Models;

public class Notification
{
    public int NotificationID { get; set; }

    public int UserID { get; set; }
    public User User { get; set; } = null!;

    public int? RelatedWorkItemID { get; set; }
    public WorkItem? RelatedWorkItem { get; set; }

    public int? RelatedSprintID { get; set; }
    public Sprint? RelatedSprint { get; set; }

    public string NotificationType { get; set; } = null!;
    public string Message { get; set; } = null!;

    public bool IsRead { get; set; }
    public DateTime? ReadAt { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: dsb-frontend/src/types/notification.ts
export type UnreadNotificationCount = {
  unreadCount: number;
};

export type NotificationListItem = {
  notificationID: number;
  notificationType: string;
  message: string;
  relatedWorkItemID?: number | null;
  relatedSprintID?: number | null;
  isRead: boolean;
  readAt?: string | null;
  createdAt: string;
};

export type NotificationListResponse = {
  page: number;
  pageSize: number;
  total: number;
  unreadCount: number;
  items: NotificationListItem[];
};